cmd/grype/cli/commands/db_search.go

```diff
@@ -164,3 +286,194 @@ func presentLegacy(outputFormat string, vulnerabilities []vulnerability.Vulnerab
+// | wolfi:distro:wolfi:rolling           |
+// | github:language:go                   |
+// | alpine:distro:alpine:3.20            |
+// | alpine:distro:alpine:3.21            |
+// | alpine:distro:alpine:edge            |
+// | github:language:rust                 |
+// | github:language:python               |
+// | sles:distro:sles:11                  |
+// | oracle:distro:oraclelinux:5          |
+// | github:language:ruby                 |
+// | github:language:dotnet               |
+// | alpine:distro:alpine:3.12            |
+// | alpine:distro:alpine:3.13            |
+// | alpine:distro:alpine:3.14            |
+// | alpine:distro:alpine:3.15            |
+// | alpine:distro:alpine:3.16            |
+// | alpine:distro:alpine:3.17            |
+// | alpine:distro:alpine:3.18            |
+// | alpine:distro:alpine:3.19            |
+// | mariner:distro:mariner:2.0           |
+// | github:language:java                 |
+// | github:language:dart                 |
+// | amazon:distro:amazonlinux:2023       |
+// | alpine:distro:alpine:3.10            |
+// | alpine:distro:alpine:3.11            |
+// | alpine:distro:alpine:3.4             |
+// | alpine:distro:alpine:3.5             |
+// | alpine:distro:alpine:3.7             |
+// | alpine:distro:alpine:3.8             |
+// | alpine:distro:alpine:3.9             |
+// | mariner:distro:azurelinux:3.0        |
+// | mariner:distro:mariner:1.0           |
+// | alpine:distro:alpine:3.3             |
+// | alpine:distro:alpine:3.6             |
+// | amazon:distro:amazonlinux:2022       |
+// | alpine:distro:alpine:3.2             |
+// | github:language:swift                |
+// +--------------------------------------+
+func v5Namespace(row dbsearch.AffectedPackage) string {
+	switch row.Vulnerability.Provider {
+	case "nvd":
+		return "nvd:cpe"
+	case "github":
+		language := row.Package.Ecosystem
+		// normalize from purl type, github ecosystem types, and vunnel mappings
+		switch strings.ToLower(row.Package.Ecosystem) {
+		case "golang", "go-module":
+			language = "go"
+		case "composer", "php-composer":
+			language = "php"
+		case "cargo", "rust-crate":
+			language = "rust"
+		case "dart-pub", "pub":
+			language = "dart"
+		case "nuget":
+			language = "dotnet"
+		case "maven":
+			language = "java"
+		case "swifturl":
+			language = "swift"
+		case "npm", "node":
+			language = "javascript"
+		case "pypi", "pip":
+			language = "python"
+		case "rubygems", "gem":
+			language = "ruby"
+		}
+		return fmt.Sprintf("github:language:%s", language)
+	}
+	if row.OS != nil {
+		family := row.OS.Name
+		switch row.OS.Name {
+		case "amazon":
+			family = "amazonlinux"
+		case "mariner":
+			switch row.OS.Version {
+			case "1.0", "2.0":
+				family = "mariner"
+			default:
+				family = "azurelinux"
+			}
+		case "oracle":
+			family = "oraclelinux"
+		}
+
+		return fmt.Sprintf("%s:distro:%s:%s", row.Vulnerability.Provider, family, row.OS.Version)
```

Comment:
I think this will return the wrong string for redhat, since the namespace has `redhat` but the provider is called `rhel`?

Reply:
resolved with the latest commit